front/components/database/DatabaseModal.tsx

```diff
@@ -0,0 +1,257 @@
+          if (res.value.content.length > 1000000) {
+            sendNotification({
+              type: "error",
+              title: "File too large",
+              description:
+                "Please upload a file containing less than 1 million characters.",
+            });
+            return;
+          }
+
+          const uploadRes = await fetch(
+            `/api/w/${workspaceId}/data_sources/${dataSource.name}/databases/csv`,
+            {
+              method: "POST",
+              body: JSON.stringify({
+                name: name,
+                description: newTableDescription,
+                csv: content,
+              }),
+              headers: {
+                "Content-Type": "application/json",
+              },
+            }
+          );
+
+          if (!uploadRes.ok) {
+            sendNotification({
+              type: "error",
+              title: "Error uploading file",
+              description: `An error occured: ${await uploadRes.text()}.`,
+            });
+            return;
+          }
+
+          await mutate(
+            `/api/w/${workspaceId}/data_sources/${dataSource.name}/databases?offset=0&limit=100`
+          );
+
+          close();
+        }}
+      />
+
+      <div className="mx-auto flex max-w-2xl flex-col">
+        <div className="w-full pt-12">
+          <div className="flex flex-col gap-4 px-2">
+            <div className="text-lg font-bold text-element-900">Name</div>
+            {database ? (
+              <div className="text-lg text-element-700">{database.name}</div>
+            ) : (
+              <Input
+                name="name"
+                placeholder="Name"
+                value={name}
+                onChange={(v) => {
+                  setName(v);
+                }}
+                className="w-full"
+                error={isNameValid() ? null : "Invalid name"}
+              />
+            )}
+            {!database && (
+              <Input
+                name="description"
+                placeholder="Description"
+                className="w-full"
+                value={newTableDescription}
+                onChange={(v) => {
+                  setNewTableDescription(v);
+                }}
+                error={newTableDescription.length === 0 ? "Required" : null}
+              />
+            )}
+          </div>
+        </div>
+        {database && tables && (
+          <div className="w-full pt-12">
+            <div className="flex flex-col gap-4 px-2">
+              <div className="text-lg font-bold text-element-900">Tables</div>
+              {tables.map((t) => {
+                const editedDescription =
+                  editedDescriptionByTableId[t.table_id];
+                return (
+                  <div
+                    className="flex flex-col gap-2 text-lg text-element-700"
+                    key={t.table_id}
+                  >
+                    {t.name}
+                    <Input
```

Comment:
What is description used for? is it used by the model in any way? What is a good description?

Reply:
It is send to the model indeed. I specified it on the field description.